code/backend/batch/utilities/tools/question_answer_tool.py

```diff
@@ -124,47 +110,52 @@ def generate_on_your_data_llm_chain(
             separators=(",", ":"),
         )
 
-        return answering_prompt, {
-            "sources": documents,
-            "question": question,
-            "chat_history": chat_history,
-        }
+        return [
+            {
+                "content": self.config.prompts.answering_system_prompt,
+                "role": "system",
+            },
+            *examples,
+            {
+                "content": self.env_helper.AZURE_OPENAI_SYSTEM_MESSAGE,
+                "role": "system",
+            },
+            *QuestionAnswerTool.clean_chat_history(chat_history),
+            {
+                "content": self.config.prompts.answering_user_prompt.format(
+                    sources=documents,
+                    question=question,
+                ),
+                "role": "user",
+            },
+        ]
 
-    def answer_question(
-        self, question: str, chat_history: list[SourceDocument], **kwargs
-    ):
+    def answer_question(self, question: str, chat_history: list[dict], **kwargs):
         source_documents = Search.get_source_documents(self.search_handler, question)
 
         if self.config.prompts.use_on_your_data_format:
-            answering_prompt, input = self.generate_on_your_data_llm_chain(
+            messages = self.generate_on_your_data_messages(
                 question, chat_history, source_documents
             )
         else:
             warnings.warn(
                 "Azure OpenAI On Your Data prompt format is recommended and should be enabled in the Admin app.",
             )
-            answering_prompt, input = self.generate_llm_chain(
-                question, source_documents
-            )
+            messages = self.generate_messages(question, source_documents)
 
         llm_helper = LLMHelper()
 
-        answer_generator = LLMChain(
-            llm=llm_helper.get_llm(), prompt=answering_prompt, verbose=self.verbose
-        )
-
-        with get_openai_callback() as cb:
-            result = answer_generator(input)
+        response = llm_helper.get_chat_completion(messages, temperature=0)
```

Comment:
Should this be:

```suggestion
        response = llm_helper.get_chat_completion(messages, temperature=self.env_helper.AZURE_OPENAI_TEMPERATURE)
```
?

Reply:
It probably should be, however I've tried to keep it the same as before. In LangChain it's hardcoded to `0`.

`AZURE_OPENAI_TEMPERATURE` is only used in the BYOD endpoint

Reply:
Gotcha, I'm thinking we should probably set that now that we have control. The env var defaults to 0 so shouldn't cause any issues.

Reply:
We had control before, we just set it to 0

https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/blob/54723fe9b13baf1096abff135c67a98fcdf1a2cb/code/backend/batch/utilities/helpers/llm_helper.py#L43

I'll keep it as 0 for now to reduce the changes from what should essentially be a refactor, but will raise a new issue for this

Reply:
#941